text/3424-cargo-script.md

```diff
@@ -194,6 +176,11 @@ $ cargo --verbose ./hello_world.rs
 Did our date match? true
 ```
 
+Cargo will cache the exact information about which revision of all of these dependencies we used.
+
+Now, if `regex` gets updated, we will still build with the same revision until
+we choose to `cargo update --manifest-path hello_world.rs`.
```

Comment:
How does this work? *Where* is this cached? Where does the equivalent of the lockfile live?

This seems a little too opaque.

Reply:
That was intentional as lockfiles are another big area for us to try to figure out the right way to handle them is.  The Pre-RFC also has [a section for this](https://github.com/epage/cargo-script-mvs/blob/main/0000-cargo-script.md#lockfile)

Reply:
I'm not suggesting the eRFC has to fully specify embedded lockfiles or any of the many other aspects discussed in the internals thread. I just think if we're going to state that this is cached and updated, we should give *some* idea of whether this is stored in the cargo home directory (which I'd expect), or next to the script (which I hope it isn't), or similar.

(Also, on a separate note, can we make just `cargo update hello_world.rs` work?)

Reply:
> I'm not suggesting the eRFC has to fully specify embedded lockfiles or any of the many other aspects discussed in the internals thread. I just think if we're going to state that this is cached and updated, we should give some idea of whether this is stored in the cargo home directory (which I'd expect), or next to the script (which I hope it isn't), or similar.

Done

> (Also, on a separate note, can we make just cargo update hello_world.rs work?)

My understanding was that cargo intentionally avoided positional arguments (`cargo add` being an exception) and one reason for this is due to the heavy use of `num_args(0..=1)` arguments like `--timings`.

In the Pre-RFC, I did bring up whether we should provide a short-flag for `--manifest-path` because of how cumbersome it is to use for these files that aren't auto-discovered. 

I'll at least call this out as another area to explore.

Reply:
@epage I just read https://github.com/epage/rfcs/blob/cargo-script/text/3424-cargo-script.md, but couldn't find a place where it mentions what happens with lock files at all. I was searching for `lock` string after reading the whole document.

Reply:
> Cargo will cache the exact information (in a location referred to as
`CARGO_HOME`) about which revision of all of these dependencies we used.

Reply:
(I didn't more explicitly call out `Cargo.lock` because, narratively within that document this is "updating", `Cargo.lock` would be meaningless to the user.  That would need to be covered later when talking about multi-file packages)

Reply:
Should we allow some way of getting the information that a Rust developer would expect from a lock file?

Reply:
What do you mean?

Reply:
> 

isn't that just `cargo metadata --manifest-path=path/to/my-script.rs`?

Reply:
Great. I didn’t use `cargo metadata` before. It will do the trick, albeit in a format that will be unfamiliar for many people.

Reply:
I looked at both `cargo metadata` and `cargo tree` to see if everything contained in a lock file is present. The one thing that I see missing is `checksum` field. I think it can be pretty useful.

Reply:
Turns out that `checksum` is internal implementation checksum useful only to cargo, and is not that useful to end users.

Reply:
One thing I don't see mentioned is whether there's only a single lockfile per system, or one per script. If there's a lockfile per script, what happens if a script is renamed? Does that reset all the dependency versions? What about moving a script file to a different directory without renaming it?

Phrased differently, the document says "Cargo will cache the exact information" but doesn't specify what key the cache is indexed by.

Reply:
That sounds like we'll need `cargo venv` lol